### ISSUE TITLE
Tuya W600-based socket

### ISSUE THREAD

**Issue by bj-tracer** [RavenSystem/esp-homekit-devices#977]
Hello!
I bought a new Tuya smart socket in new form factro and with energy monitoring.
Like this https://aliexpress.ru/item/4000478798085.html
But it's based on esp-compatible module with WinnerMicro W600 controller (http://www.winnermicro.com/en/html/1/156/158/497.html).

Can it flashing with HAA?

**Comment by tonysprenk** [comment]
No, maybe it’s possible to replace the module with an ESP8266XX module. Can you post pictures of the module inside?

**Comment by bj-tracer** [comment]
Hi!
I haven’t deep disassembling of it now.
Only this pictures:
![A4C274F3-3E1E-4007-986D-2A512F60BFC1](https://user-images.githubusercontent.com/50882159/83292078-2b5e9d80-a1f2-11ea-95b6-d96b8c5f0018.jpeg)
![50DD8DEE-574C-4533-82DB-66F284337893](https://user-images.githubusercontent.com/50882159/83292092-30235180-a1f2-11ea-8404-9ddb4a874f01.jpeg)

Module docs: https://docs.w600.fun/?p=product/tw-02.md

**Comment by tonysprenk** [comment]
It’s a winnermicro w600 chip. As far as I know there is no esp8266 equivalent to this module. What you can do is get an esp01 or similar and connect it to the 3.3v, gnd and gpio pins with small wires. Probably not worth the hassle though.

**Comment by bj-tracer** [comment]
I know it.
But people flash it with Tasmota firmware by OTA as I know.

**Comment by bj-tracer** [comment]
Ok
I replaced module to esp-M3 and flashed HAA.
It's work! :)

**Comment by tonysprenk** [comment]
Nice! Do you have any pictures of the process? Or before and after pictures?

**Comment by bj-tracer** [comment]
It is the first sample. I tried to use grio2 to control the relay. But it’s not working.

 ![411F2673-C90F-4DF9-A13F-62BEAA19B9A4](https://user-images.githubusercontent.com/50882159/83968127-b36a3480-a8cf-11ea-94c3-e7ad0049fda8.jpeg)
![A751B4BE-1366-4B71-9F28-75999AB1ABE7](https://user-images.githubusercontent.com/50882159/83968130-b6fdbb80-a8cf-11ea-8231-d5ad05f664bd.jpeg)
![744464E8-6B04-440E-BA01-3D4694EECC02](https://user-images.githubusercontent.com/50882159/83968132-b7965200-a8cf-11ea-9790-32c3a344e1af.jpeg)
![DD05C630-51A6-4361-B536-844D5ED452C5](https://user-images.githubusercontent.com/50882159/83968134-b82ee880-a8cf-11ea-88af-e6a1a8bc228e.jpeg)

And I changed gpio2 to gpio12.
